src/storage/mod.rs

```diff
@@ -938,15 +937,20 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
         &self,
         mut ctx: Context,
         max_ts: TimeStamp,
-        start_key: Option<Key>,
-        end_key: Option<Key>,
+        raw_start_key: Option<Vec<u8>>,
+        raw_end_key: Option<Vec<u8>>,
```

Comment:
We can accept `Vec<u8>` and call `Key::from_raw_maybe_unbounded` here.

Reply:
I check unbou and transform to `Option<Vec<u8>>` in [`service/kv.rs`](https://github.com/pingyu/tikv/blob/98d7c823b97a83cbc1f823f5b3cef3966b190b27/src/server/service/kv.rs#L1424-L1433). So just `Key::from_raw` is needed here.

Reply:
I mean by checking it here, we save repeating the logic of `Key::from_raw_maybe_unbounded` in kv.rs and also no long bother to `raw_start_key.as_ref().map(|x| Key::from_raw(&x[..]))` later on.

Reply:
But if so, we need to check `is_empty()` again before passing to `check_api_version()`.

Besides, the signature as `Option` telling that the argument is optional for unbounded. I think it's more readable.